feat: update password page

oopshell opened the pull request:
## Main feature

#### Added user password reset page:
- add account management to sidebar
- create account management page
- add change password card in account management page as clickable card, navigating to change password page 
- create change password page
- create change password form
- create shared component: styled modal for: showing user password reset success in dialogue
- add password reset success modal after updating password successfully, navigating to login page after clicking "Ready to trade again" button on modal
- add gql mutation of changing password
- add route config for account management and change password page

#### Added testing:
- unit test for account management page
- unit test for card link component
- unit test for change password page

#### Resolve CP-34

---

- Entry to account management page
<img width="200" alt="Screenshot 2024-06-13 at 11 35 28" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/9fe73a2e-e05b-4b3f-a36a-121e30ef7e21">
<img width="200" alt="Screenshot 2024-06-13 at 11 35 59" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/2502c64b-db07-4a22-b140-a4e4cd322ba3">

- Account management page
<img width="720" alt="Screenshot 2024-06-13 at 11 38 38" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/7c7320bf-cb3c-446f-ba9e-71120770f52e">

- Reset password page
<img width="720" alt="Screenshot 2024-06-13 at 11 39 57" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/d8412108-112c-4092-8cc0-b776c1e5d788">

- Password criteria hint
<img width="720" alt="Screenshot 2024-06-13 at 11 40 16" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/f1cb4325-f847-4afa-b5c4-9070f561bd72">

- Repeat password not match alert
<img width="720" alt="Screenshot 2024-06-13 at 11 41 01" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/e7ce0a21-afe7-434d-a1c2-ab0211776277">

- Current password incorrect alert
<img width="720" alt="Screenshot 2024-06-13 at 11 41 46" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/94ebd291-ee7a-4920-ab4b-ede4f151caf7">

- Password reset success message
<img width="720" alt="Screenshot 2024-06-13 at 11 42 04" src="https://github.com/BeeQuantAI/platform_app/assets/48723564/51a20a7d-9d50-4311-96aa-77d026ebe230">

Chris-Wang commented on src/routes/routeConfig.ts:
```diff
@@ -86,6 +90,18 @@ export const ROUTE_CONFIG: Record<string, IRoute> = {
+    component: AccountManagement,
+  },
+  [ROUTE_KEY.CHANGE_PASSWORD]: {
+    path: '/account/change-password',
```
consider using `account/update/password` for path

oopshell replied:
![image](https://github.com/BeeQuantAI/platform_app/assets/48723564/f166490d-8950-4421-b79c-f192117769f0)

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
```diff
@@ -0,0 +1,142 @@
+import { Button } from '@/shared/components/Button';
+import { Controller, useForm } from 'react-hook-form';
+import ChangePasswordSuccess from './ChangePasswordSuccess';
+// import { passwordPatten } from '@/shared/utils/helpers';
```
remove commented code in pr, unless attached reasons

oopshell replied:
![image](https://github.com/BeeQuantAI/platform_app/assets/48723564/5f163ce8-d7de-4309-814e-dd50444343d5)

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
```diff
@@ -0,0 +1,142 @@
+import { Controller, useForm } from 'react-hook-form';
+import ChangePasswordSuccess from './ChangePasswordSuccess';
+// import { passwordPatten } from '@/shared/utils/helpers';
+const passwordPatten = /^(?=.*[A-Za-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-])[A-Za-z0-9#?!@$%^&*-]{8,32}$/;
```
use regex from helper file, Ray made a good one

oopshell replied:
![image](https://github.com/BeeQuantAI/platform_app/assets/48723564/77383039-7a71-44b3-b73c-87ad2466deed)

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
```diff
@@ -0,0 +1,142 @@
+                <Alert variant="danger" show={!!error}>
+                  {error}
+                </Alert>
+
```
why there is empty line

oopshell replied:
Removed

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
```diff
@@ -0,0 +1,142 @@
+                    />
+                  </FormGroupField>
+                </FormGroup>
+
```
why there is empty line

oopshell replied:
Removed

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
```diff
@@ -0,0 +1,142 @@
+                    />
+                  </FormGroupField>
+                </FormGroup>
+
```
why there is empty line

oopshell replied:
Removed

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
```diff
@@ -0,0 +1,142 @@
+                    />
+                  </FormGroupField>
+                </FormGroup>
+
```
why there is empty line

oopshell replied:
Removed

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/index.ts:
```diff
@@ -0,0 +1 @@
+export { default } from './ChangePassword';
```
consider moving ChangePassword directly under `Account`

oopshell replied:
![image](https://github.com/BeeQuantAI/platform_app/assets/48723564/5cd1b068-aefb-4d13-8fc6-a416ebf19250)

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/components/ChangePasswordForm.tsx:
Consider to add password format hint as we did on Register page, introduce `onFocus` and `onBlur`

oopshell replied:
Added onFocus and onBlur control
Added hint
![image](https://github.com/BeeQuantAI/platform_app/assets/48723564/0f398f38-c9a4-4988-8d91-16703ded17ba)

Chris-Wang commented on src/containers/Account/AccountManagement/components/ChangePassword/ChangePassword.test.tsx:
use it('should ...') for each test

oopshell replied:
Modified

Chris-Wang commented on jest.config.ts:
```diff
@@ -6,6 +6,11 @@ module.exports = {
     '\\.(css|less)$': '<rootDir>/__mocks__/styleMock.ts',
     '\\.(gif|ttf|eot|svg)$': '<rootDir>/__mocks__/fileMock.ts',
+    '^styled-theming$': '<rootDir>/__mocks__/styledThemingMock.ts',
+    '^mdi-react/EyeIcon$': '<rootDir>/__mocks__/EyeIconMock.tsx',
```
if some icon is only used on one page, it shouldn't be introduced globally. I suggest mock it in each test file.

oopshell replied:
Removed mocks from global

Chris-Wang commented:
Need to merge api part firstly

Chris-Wang reviewed (changes requested):
Need to update sideBar and topBar to add update component,
commits have duplicated text,
please merge the latest dev to your branch

Chris-Wang reviewed (changes requested):
Still there're no access to the new component from ui

oopshell commented:
> Still there're no access to the new component from ui

Ya, I see the problem, and working on it. The backend data input schema changes, so I need to update several files in the frontend. I will fix this as soon as possible. Also, after fixing problems, I will add entry of the page to sidebar and topbar.

SeijaHuang commented:
Feat: add update password page

Added update password link in the account management page
- Added unit tests for update password link
Added update password page
- Implemented update password form
- Added unit tests for update password page
Modularized update password link
- Added protected/account/management fold to contain all components related to update password link
Modularized update password page
- Added protected/account/update fold to contain all components related to update password page
- Refactored update password form in alignment with login-form and setting-form modularized before
- Refactored the test file to update codes with new refactored form
Fixed the conflicts between CP-34 with dev

Resolve CP-34

SeijaHuang committed:
Feat: add update password page

Added update password link in the account management page
- Added unit tests for update password link
Added update password page
- Implemented update password form
- Added unit tests for update password page
Modularized update password link
- Added protected/account/management fold to contain all components related to update password link
Modularized update password page
- Added protected/account/update fold to contain all components related to update password page
- Refactor update password form in alignment with login-form and setting-form modularized before
- Refactor the test file to update codes with new refactored form
- Fixed the conflicts between CP-34 with dev

Refactor shared component
- move FormIcon, FormInput, and FormPasswordInput into shared component fold to make it shareable between all the forms, including setting page form and update password form

Resolve CP-34
